### PR TITLE
initial setup

### PR DESCRIPTION
**Pull request by hjdivad**
- Add simple configuration
- Setup CI
- Configure LSP globals

TODO:
- [x] fix CI setup (download nvim as in hjdivad/dotfiles)
- [x] fix test run in CI

---

> Right now there are lots of CI failures like [this one](https://github.com/malleatus/common.nvim/runs/6647216110?check_suite_focus=true) where
> 
> ```bash
> make --version
> make -f Makefile.ci test
> # => Error detected while processing command line:
> ```
> 
> And then times out.
> 
> If I ssh to the box (with `DebugSSH true`), then `make -f Makefile.ci test` passes fine.  It's almost identical to the setup in https://github.com/hjdivad/dotfiles/pull/56.
> 

This looks to be the result of quitting via the command line rather than as an autocommand on `PaqDoneInstall`

### CODE REVIEW

**Comment by hjdivad** [comment]
@rwjblue nothing too exciting but enough to get going.

see https://github.com/hjdivad/dotfiles/commit/5206baf6258d80d46e003a8ce3ea6f5f560266f6 for installation + local linking.

obviously a better local link would be nice but this is at least doable.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+green := $(shell tput setaf 2)
+yellow := $(shell tput setaf 3)
+blue := $(shell tput setaf 4)
+cyan := $(shell tput setaf 6)
+cyan80 := $(shell tput setaf 86)
+grey500 := $(shell tput setaf 244)
+grey300 := $(shell tput setaf 240)
+bold := $(shell tput bold)
+underline := $(shell tput smul)
+reset := $(shell tput sgr0)
```
Comment (hjdivad) [inline Makefile:21]: this (`tput`) is apparently CI-unfriendly but we can probably find some straightforward "if not interactive bail" make snippet

**Comment by hjdivad** [comment]
@hjdivad reminder to post a relevant `.vimrc.lua` for setting up `<leader>rt`

**Review by rwjblue — APPROVED** [review]

File: tests/malleatus/utils_spec.lua [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      assert.equals(true, vim.o.smartcase)
+    end)
+
+    it('opts in to ftdetect.lua', function()
+      malleatus.setup()
+      assert.equals(1, vim.g.do_filetype_lua)
+      assert.equals(0, vim.g.did_load_filetypes)
+    end)
+  end)
+end)
```
Comment (hjdivad) [inline tests/malleatus/utils_spec.lua:25]: Recommended to have 

```lua
-- .vim/ftplugin/lua.lua

local filename = vim.fn['expand']('%:t')

if filename:find('_spec.lua$') then
  vim.cmd('UltiSnipsAddFiletypes plenary')

  -- vim-test doesn't support Plenary
  vim.keymap.set('n', '<leader>rt', '<Plug>PlenaryTestFile', { buffer = true })
end

```

**Pull request merged**